camelot/core/orm/entity.py

```diff
@@ -268,9 +272,49 @@ def __new__( cls, classname, bases, dict_ ):
                     break
             else:
                 dict_.setdefault('__mapper_args__', dict())
-
-        return super( EntityMeta, cls ).__new__( cls, classname, bases, dict_ )
-
+            
+            # Initialize the types that are allowed registering classes for as None if not set.
+            for base in bases:
+                if hasattr(base, '__types__'):
+                    break
+            else:
+                dict_.setdefault('__types__', None)
+                # Dict that stores class-type registrations. 
+            
+            types = dict_.get('__types__')
+            if types is not None:
+                assert isinstance(types, util.OrderedProperties), 'The set type should be an instance of sqlalchemy.util.OrderedProperties.'
+                dict_.setdefault('_cls_for_type', dict())
+                    
+        _class = super( EntityMeta, cls ).__new__( cls, classname, bases, dict_ )
+        cls.register_class(cls, _class, dict_)
+        return _class
+        
+    def register_class(cls, _class, dict_):
```

Comment:
zet het gebruik van die __for_type__ class attribute aub in de documentatie, en ook hoe je dan terug aan je class komt, want dat is voor de casual reader onbegrijpelijk :) + noteer wat het verschil is met inheritance

Reply:
Ok, zie e6a5b8057ea789c266e538befbdb3e31a1e5efab.